### ISSUE TITLE
bug identified in development branch

### ISSUE THREAD

**Issue by feiranl** [SysBioChalmers/Human-GEM#628]
import and export cycle gives an error:
```
model = importYaml('Human-GEM.yml');
>> exportYaml(model, 'Human-GEM.yml');
Index exceeds the number of array elements. Index must not exceed 13087.

Error in exportYaml>writeField (line 190)
            list = field{pos};  %subSystems already comes in a cell array

Error in exportYaml (line 86)
    writeField(model, fid, 'subSystems',           'txt', i, '  - subsystem')
```

This is due to the subsystems length is different with other rxn related field.
<img width="358" alt="Screenshot 2023-06-12 at 10 51 09 AM" src="https://github.com/SysBioChalmers/Human-GEM/assets/32157802/64ad23a3-84c6-4113-a224-f7c25dea4535">

**Comment by mihai-sysbio** [comment]
Oh that's surprising to see. There is a GH workflow that runs the [yaml conversion check](https://github.com/SysBioChalmers/Human-GEM/blob/main/code/test/testYamlConversion.m). It looks like when the test fails the workflow still passes - have a look at the step _Run conversion script_ for [this workflow ron](https://github.com/SysBioChalmers/Human-GEM/actions/runs/5237189388/jobs/9455204803) on the merged PR. Looking back through the results, the problem might have been introduced in https://github.com/SysBioChalmers/Human-GEM/pull/606.

**Comment by mihai-sysbio** [comment]
My suggestion is that the fix should also include a change in the test such that when it fails the action also fails. After this, the fix of the yaml file should be committed, and we should see the workflow pass.

**Comment by feiranl** [comment]
it errors in the check, but didnot fail in the [action](https://github.com/SysBioChalmers/Human-GEM/pull/606/checks):


                                           

 
{Index exceeds the number of array elements (13087).

Error in exportYaml>writeField (line 190)
            list = field{pos};  %subSystems already comes in a cell array

Error in exportYaml (line 86)
    writeField(model, fid, 'subSystems',           'txt', i, '  - subsystem')

Error in testYamlConversion (line 25)
exportYaml(model,'testYamlConversion.yml');

Error in run (line 91)
evalin('caller', strcat(script, ';'));
} 

@haowang-bioinfo maybe you can fix this?

**Comment by haowang-bioinfo** [comment]
> @haowang-bioinfo maybe you can fix this?

yes, working on this

**Comment by haowang-bioinfo** [comment]
> Looking back through the results, the problem might have been introduced in https://github.com/SysBioChalmers/Human-GEM/pull/606.

the error appears to be introduced in #619, instead of #606

**Comment by haowang-bioinfo** [comment]
@feiranl the error encountered during yaml import and export cycle is now fixed by commit 182010dd5e6ae1a3441f5672eaecc4bf635afa68 in #629

**Comment by mihai-sysbio** [comment]
Based on the merged pull requests I will close this issue, please re-open if needed.
